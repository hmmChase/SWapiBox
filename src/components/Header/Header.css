.Header {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  justify-content: space-between;
  align-items: center;
  padding: 0 50px 0 50px;
  border-bottom: 2px solid yellow;
  background-image: url(https://thumbs.gfycat.com/IgnorantDimpledHyrax-size_restricted.gif);
  background-repeat: no-repeat;
  background-size: cover;
}

.title {
  color: #000;
  font-family: 'Audiowide', sans-serif;
  -webkit-text-stroke: 1px yellow;
}

.categories {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0 2em 0 2em;
}

.num-favs {
  margin-left: 5px;
}

.navlink {
  padding: 10px;
  border: 2px solid yellow;
  border-radius: 10px;
  margin: 0.5em;
  background: none;
  color: #fff;
  font-size: 1.5em;
  font-weight: 600;
  text-decoration: none;
  cursor: pointer;
}

.navlink:hover {
  background: rgba(164, 164, 0, 0.5);
}

.active {
  color: #feda4a;
}
