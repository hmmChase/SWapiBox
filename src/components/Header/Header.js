import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';

const Header = () => {
  return (
    <section className="Header">
      <h1>SWapiBox</h1>
      <div className="categories">
        <NavLink className="navlink" to="/people">
          People
        </NavLink>
        <NavLink className="navlink" to="/planets">
          Planets
        </NavLink>
        <NavLink className="navlink" to="/vehicles">
          Vehicles
        </NavLink>
      </div>

      <NavLink className="navlink" to="/favorites">
        Favorites
        {/* <span>{props.favorites}</span> */}
      </NavLink>
    </section>
  );
};

export default Header;
