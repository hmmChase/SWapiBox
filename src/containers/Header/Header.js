import React from 'react';
import './Header.css';

const Header = () => {
  return <section className="Header">Header</section>;
};

export default Header;
