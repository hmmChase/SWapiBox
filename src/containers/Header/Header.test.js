import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  let header;

  beforeEach(() => {
    header = shallow(<Header />);
  });

  it('matches the snapshot', () => {
    expect(header).toMatchSnapshot();
  });
});
