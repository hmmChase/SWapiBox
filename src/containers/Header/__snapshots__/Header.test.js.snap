// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          SWapiBox
</h1>,
        <nav>
          <NavButton
                    category="people"
                    setCategory={undefined}
          />
          <NavButton
                    category="planets"
                    setCategory={undefined}
          />
          <NavButton
                    category="vehicles"
                    setCategory={undefined}
          />
          <Favorites
                    category="favorites"
                    favorites={undefined}
                    setFavorites={undefined}
          />
</nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWapiBox",
        },
        "ref": null,
        "rendered": "SWapiBox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavButton
              category="people"
              setCategory={undefined}
/>,
            <NavButton
              category="planets"
              setCategory={undefined}
/>,
            <NavButton
              category="vehicles"
              setCategory={undefined}
/>,
            <Favorites
              category="favorites"
              favorites={undefined}
              setFavorites={undefined}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "category": "people",
              "setCategory": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "category": "planets",
              "setCategory": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "category": "vehicles",
              "setCategory": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "category": "favorites",
              "favorites": undefined,
              "setFavorites": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SWapiBox
</h1>,
          <nav>
            <NavButton
                        category="people"
                        setCategory={undefined}
            />
            <NavButton
                        category="planets"
                        setCategory={undefined}
            />
            <NavButton
                        category="vehicles"
                        setCategory={undefined}
            />
            <Favorites
                        category="favorites"
                        favorites={undefined}
                        setFavorites={undefined}
            />
</nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWapiBox",
          },
          "ref": null,
          "rendered": "SWapiBox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavButton
                category="people"
                setCategory={undefined}
/>,
              <NavButton
                category="planets"
                setCategory={undefined}
/>,
              <NavButton
                category="vehicles"
                setCategory={undefined}
/>,
              <Favorites
                category="favorites"
                favorites={undefined}
                setFavorites={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "category": "people",
                "setCategory": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "category": "planets",
                "setCategory": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "category": "vehicles",
                "setCategory": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "category": "favorites",
                "favorites": undefined,
                "setFavorites": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
